src/apps/activity_assignments/crud/assignments.py

```diff
@@ -111,6 +111,44 @@ async def already_exists(self, schema: ActivityAssigmentSchema) -> ActivityAssig
         db_result = await self._execute(query)
         return db_result.scalars().first()
 
+    async def get_target_subject_ids_by_activity_or_flow_ids(
+        self,
+        respondent_subject_id: uuid.UUID,
+        activity_or_flow_ids: list[uuid.UUID] = [],
+    ) -> list[uuid.UUID]:
+        """
+        Retrieves the IDs of target subjects that have assignments to be completed by the provided respondent.
+
+        Parameters:
+        ----------
+        respondent_subject_id : uuid.UUID
+            The ID of the respondent subject to search for. This parameter is required.
+        activity_or_flow_ids : list[uuid.UUID]
+            Optional list of activity or flow IDs to narrow the search. These IDs may correspond to either
+            `activity_id` or `activity_flow_id` fields
+
+        Returns:
+        -------
+        list[uuid.UUID]
+            List of target subject IDs associated with the provided activity or flow IDs.
+        """
+        query = select(ActivityAssigmentSchema.target_subject_id).where(
+            ActivityAssigmentSchema.respondent_subject_id == respondent_subject_id,
+            ActivityAssigmentSchema.soft_exists(),
```

Comment:
I think we should probably some JOINs with the subjects, since assignments are currently not soft-deleted when subjects are deleted (which I personally think is a data integrity issue – I think assignments should be getting soft-deleted in a cascade fashion, but for whatever reason, they aren't yet). And since this is being done in the `get_by_applet` function below, I think it probably makes sense to include here:
```py
        respondent_schema = aliased(SubjectSchema)
        target_schema = aliased(SubjectSchema)

        query = (
            …
            .join(respondent_schema, respondent_schema.id == ActivityAssigmentSchema.respondent_subject_id)
            .join(target_schema, target_schema.id == ActivityAssigmentSchema.target_subject_id)
            …
            .where(
                …
                respondent_schema.soft_exists(),
                target_schema.soft_exists(),
```

Reply:
I think that's accounted for here

https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/3188749001144cf97ec3025493885a77c9b2c42f/src/apps/subjects/api.py#L354

The `get_by_ids` function has a parameter called `include_deleted` that defaults to `False`

Reply:
Oh awesome, that keeps things simpler. Good catch.